plugin-manager/src/main/kotlin/com/itsaky/androidide/plugins/manager/services/IdeProjectServiceImpl.kt

```
@@ -80,6 +81,19 @@ class IdeProjectServiceImpl(
         }
     }
 
+    override fun getModuleContext(filePath: String): ModuleContext? {
+        if (!hasRequiredPermissions()) {
+            throw SecurityException("Plugin $pluginId does not have required permissions: ${getRequiredPermissionsString()}")
+        }
+
+        return try {
+            ModuleContextResolver.resolve(filePath)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)